From b24745479ce65720d88f6ea27259771a209157e7 Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Tue, 7 Nov 2023 09:05:17 -0600
Subject: [PATCH] Only add the GL dependency as a requires when it is a
 pkgconfig dep

On macOS, this solves a problem when using the system's GL library.
Use the OpenGL framework on macOS.
---
 meson.build | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 7949e30..24fa8fb 100644
--- a/meson.build
+++ b/meson.build
@@ -18,7 +18,12 @@ gl_provider = get_option('gl_provider')
 if gl_provider == 'glvnd'
   gl_provider = 'opengl'
 endif
-dep_gl = dependency(gl_provider)
+
+if host_machine.system() == 'darwin'
+  gl_dep = dependency('appleframeworks', modules : ['OpenGL'])
+else
+  dep_gl = dependency(gl_provider)
+endif
 
 inc_include = include_directories('include')
 
@@ -34,5 +39,5 @@ pkgconf.generate(
   libglu,
   name : 'glu',
   description : 'Mesa OpenGL Utility Library',
-  requires: dep_gl
+  requires: dep_gl.type_name() == 'pkgconfig' ? dep_gl : []
 )
-- 
2.41.0

